### PR TITLE
New header

### PR DESCRIPTION
**Pull request by cozmo**
Simplifies and uses the new design for the header component

**Testing:**
- Manual tests:
  - [x] Chrome
  - [x] Safari
  - [ ] IE10

**Roll Out:**
- Before merging:
  - [x] Updated docs
  - [x] Bumped version in `package.json`
    - Breaking change? Run `npm version major`
    - New component or backward-compatible component feature change? Run `npm version minor`
    - Only changing documentation? All good. Skip this step.
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)

### CODE REVIEW

File: src/Logo/index.ts [hunk 1]
```diff
@@ -1,4 +0,0 @@
-import Logo from "./Logo";
-import BetaTag from "./BetaTag";
-export default Logo;
-export { BetaTag };
```
Comment (brollin) [inline src/Logo/index.ts]: I kind of like the `index.ts` file imports the other named files! Don't want to start a holy war here, just sayin' :P

File: src/TopBar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  children: PropTypes.node,
+  className: PropTypes.string,
+  logoHref: PropTypes.string.isRequired,
+  title: PropTypes.node,
+};
+
+/**
+ * Global page-level header component.
+ */
+export default class TopBar extends React.PureComponent {
```
Comment (brollin) [inline src/TopBar/index.tsx]: Can we make this a named export? The explicit nature of named exports make for an easier development experience IMO, and it's the pattern we've been moving towards elsewhere. (also applies to other components in this PR)

File: src/TopBar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        <TopBarButton href={this.props.logoHref} className="dewey-TopBar--logoLink">
+          <Logo className="dewey--TopBar--logo" />
+        </TopBarButton>
+        {title && (
+          <h1 className="dewey--TopBar--title" title={title}>
+            {title}
+          </h1>
+        )}
+        <FlexBox alignItems="center" justify={Justify.END} grow>
+          {children}
```
[the remark this answers is not in the document]
Comment (cozmo) [reply src/TopBar/index.tsx]: Consumer will handle I'm thinking.

File: src/TopBar/index.tsx (hunk 2 above)
Comment (cozmo) [reply src/TopBar/index.tsx]: I agree, was just trying to follow convention. I'll happily switch - Glad we're moving in that direction.

File: src/TopBar/index.less [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+.dewey--TopBar--title {
+  .padding--none;
+  .margin--y--none;
+  .margin--x--xs;
+  .text--medium;
+  .text--regular;
+  color: @neutral_white;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
```
[the remark this answers is not in the document]
Comment (cozmo) [reply src/TopBar/index.less]: Nice!

File: src/TopBar/TopBarMenu.tsx [hunk 5]
```diff
@@ -2,53 +2,36 @@ import * as _ from "lodash";
...
-  label: PropTypes.node.isRequired,
-};
-
-const cssClass = {
-  CARET: "DropdownButton--Caret",
-  CONTAINER: "TopBarMenu",
-  TRIGGER_CONTENT: "TopBarMenu--trigger--content",
-  TRIGGER_LABEL: "TopBarMenu--trigger--label",
-  TRIGGER: "TopBarMenu--trigger",
+  icon: PropTypes.string.isRequired,
```
[the remark this answers is not in the document]
Comment (cozmo) [reply src/TopBar/TopBarMenu.tsx]: Fixed in 3096559

File: src/Logo/index.ts (hunk 1 above)
Comment (cozmo) [reply src/Logo/index.ts]: Womp

**Pull request merged**